### PR TITLE
[FIX] l10n_it_edi: prevent generation of XML if document type is not set

### PR DESCRIPTION
**Pull request by kawkb**
# Issue:
When creating a vendor bill for a non EU partner, setting the tax to "22% G RC (Goods)" and generating the XML file, checking the generated XML file gives the error: "Tipo Documento must not be empty."

# Steps to reproduce:
- create a vendor bill with a non EU partner
- add a product with tax "22% G RC (Goods)"
- post the bill
- click Send Tax Integration
- check the generated XML file using an online checker like www.fatturacheck.it
- it will show the error "Tipo Documento must not be empty."

# Solution:
- The document type is usually computed. However, in this specific case, it cannot be computed because it does not meet any of the conditions in the computation logic. Additionally, there is no safeguard to prevent the generation of the XML file if the document type is not set.
- A check has been added to prevent XML generation when the document type is not set, ensuring the document type is always correctly provided before XML generation.

opw-4182531

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_it_edi/models/account_move.py [hunk 1]
```diff
@@ -1239,6 +1239,9 @@ def build_error(message, records):
             if moves := pa_moves.filtered(lambda move: not move.l10n_it_origin_document_type and move.l10n_it_cig and move.l10n_it_cup):
                 message = _("CIG/CUP fields of partner(s) are present, please fill out Origin Document Type field in the Electronic Invoicing tab.")
                 errors['move_missing_origin_document_field'] = build_error(message=message, records=moves)
+        if moves := self.filtered(lambda move: not move.l10n_it_document_type):
+            message = _("please fill out the Document Type field in the Electronic Invoicing tab.")
```
Comment (nle-odoo) [inline addons/l10n_it_edi/models/account_move.py:1243]: maybe this should begin with capital letter (to match other messages). and before merging the sentence should be added to `addons/l10n_it_edi/models/l10n_it_edi.pot`

**Review by nle-odoo — APPROVED** [review]
seems to makes sense to me

**Review by aboo-odoo — COMMENTED** [review]
LGTM! @clbr-odoo

**Review by clbr-odoo — APPROVED** [review]
Thanks !
@robodoo r+

**Comment by aboo-odoo** [comment]
@kawkb You need to adapt the two test failing to include your newly added error :+1:

**Comment by nle-odoo** [comment]
robodoo delegate+

**Pull request closed without merging**